app/authProvider.js

```diff
@@ -55,13 +56,32 @@ export class AuthProvider {
     app.use(this.passport.session());
   }
 
-  authenticate(options) {
+  login() {
     return (req, res, next) => {
+      const options = {
+        successReturnToOrRedirect: '/',
+        state: url.parse(req.headers.referer).pathname,
+      };
       this.passport.authenticate('oidc', options)(req, res, next);
     };
   }
 
+  loginCallback() {
+    return (req, res, next) => {
+      const redirectUrl = req.query.state;
+      const optionsWithUrl = {
+        callback: true,
+        failureRedirect: '/',
+        successReturnToOrRedirect: redirectUrl,
+      };
+
+      this.passport.authenticate('oidc', optionsWithUrl)(req, res, next);
+    };
+  }
+
   logout() {
+    this.loginRedirectUrl = null;
```

Comment:
Does this have any consequences with multiple browser sessions?

Reply:
nope - deleted